database/rrdcalc.c

```diff
@@ -407,11 +552,13 @@ void rrdcalc_free(RRDCALC *rc) {
     freez(rc->chart);
     freez(rc->family);
     freez(rc->dimensions);
+    freez(rc->foreachdim);
     freez(rc->exec);
     freez(rc->recipient);
     freez(rc->source);
     freez(rc->units);
     freez(rc->info);
+    freez(rc->spdim);
```

Comment:
Sould be `simple_pattern_free(rc->spdim);`

Reply:
No memory leaks after applying it.

Reply:
It is fixed now.